### PR TITLE
Add ResNet V2

### PR DESCRIPTION
**Pull request by vfdev-5**
A port of ResNetV2 from [mxnet.gluon](https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/model_zoo/vision/resnet.py) based on ["Identity Mappings in Deep Residual Networks"](https://arxiv.org/abs/1603.05027) paper.

- Concerning the code structure, I can merge both resnets in a single file if needed. 

- I did not ported weights as I remember that it is preferable to retrain models from scratch.

Graph visualization can be done like [this](https://gist.github.com/vfdev-5/5caf07a48bf9311d95f04db63a870d92) (visualization is better on Google Chrome).

What do you think on need to add ResNetV2 (trained) to `torchvision`?
Thanks

### CODE REVIEW

File: torchvision/models/resnet_v2.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        block (Module) : class for the residual block. Options are BasicBlockV1, BottleneckV1.
+        layers (list of int) : numbers of layers in each block
+        num_classes (int) :, default 1000, number of classification classes.
+    """
+    def __init__(self, block, layers, num_classes=1000):
+        super(ResNetV2, self).__init__()
+        assert block in (BottleneckV2, BasicBlockV2), "Argument block should be BottleneckV2 or BasicBlockV2"
+        self.inplanes = 64
+
+        self.bn0 = nn.BatchNorm2d(3, affine=False, track_running_stats=False)
```
Comment (ppwwyyxx) [inline torchvision/models/resnet_v2.py]: I assume this is a hack to handle preprocessing when you migrate an existing model to pytorch?
For the purpose of re-implementing a model in pytorch you probably should remove it.

Comment (vfdev-5) [reply torchvision/models/resnet_v2.py]: @ppwwyyxx actually this batchnorm is to be coherent with [mxnet.gluon one](https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/model_zoo/vision/resnet.py#L308) but probably the reason is what you say. In the original torch [implementation](https://github.com/facebook/fb.resnet.torch/blob/master/models/preresnet.lua#L146) there is no BN. So, yes, probably I should remove it.

**Comment by zhou13** [comment]
Any update on this?
